tck/methods/account.ts

```diff
@@ -0,0 +1,101 @@
+import {
+    AccountCreateTransaction,
+    Hbar,
+    AccountId,
+    TransactionResponse,
+    TransactionReceipt,
+} from "@hashgraph/sdk";
+
+import { sdk } from "../sdk_data";
+import { getKeyFromString } from "../utils/key";
+import { CreateAccountResponse } from "../response/account";
+
+import { CreateAccountParams } from "../params/account";
+import { applyCommonTransactionParams } from "../params/common-tx-params";
+
+export const createAccount = async ({
+    key,
+    initialBalance,
+    receiverSignatureRequired,
+    maxAutoTokenAssociations,
+    commonTransactionParams,
+    stakedAccountId,
+    stakedNodeId,
+    declineStakingReward,
+    memo,
+    autoRenewPeriod,
+    alias,
+}: CreateAccountParams): Promise<CreateAccountResponse> => {
+    let transaction = new AccountCreateTransaction().setGrpcDeadline(30000);
+
+    if (key != null) {
+        transaction.setKey(getKeyFromString(key));
+    }
+
+    if (initialBalance != null) {
+        transaction.setInitialBalance(Hbar.fromTinybars(initialBalance));
+    }
+
+    if (receiverSignatureRequired != null) {
+        transaction.setReceiverSignatureRequired(receiverSignatureRequired);
+    }
+
+    if (maxAutoTokenAssociations != null) {
+        transaction.setMaxAutomaticTokenAssociations(maxAutoTokenAssociations);
+    }
+
+    if (stakedAccountId != null) {
+        const accountId = AccountId.fromString(stakedAccountId);
+
+        transaction.setStakedAccountId(accountId);
+    }
+
+    if (stakedNodeId != null) {
+        transaction.setStakedNodeId(stakedNodeId);
+    }
+
+    if (declineStakingReward != null) {
+        transaction.setDeclineStakingReward(declineStakingReward);
+    }
+
+    if (memo != null) {
+        transaction.setAccountMemo(memo);
+    }
+
+    if (autoRenewPeriod != null) {
+        transaction.setAutoRenewPeriod(autoRenewPeriod);
+    }
+
+    if (alias != null) {
+        transaction.setAlias(alias);
+    }
+
+    if (commonTransactionParams != null) {
+        applyCommonTransactionParams(
+            commonTransactionParams,
+            transaction,
+            sdk.getClient(),
+        );
+    }
+
+    // Sign the transaction with the client operator private key if not already signed and submit to a Hedera network
+    let txResponse: TransactionResponse;
+    try {
+        txResponse = await transaction.execute(sdk.getClient());
+    } catch (error) {
+        throw new Error(error);

```

Comment:
if you throw the error here, add some comment like "error occurred during transaction execute..." or something like that. Same for receipt.

Reply:
Or just don't catch it at all, i think it would be caught in the middleware.

Reply:
Yeah, it's being caught in the middleware. Most likely I left this try/catch block in the code while debugging something earlier.